### PR TITLE
Consensus changes merge target

### PR DESCRIPTION
**Pull request by systemzax**
Resolved merge conflicts related to consensus changes : 

https://github.com/AntelopeIO/spring/pull/389
https://github.com/AntelopeIO/spring/pull/391
https://github.com/AntelopeIO/spring/pull/392

Re-enabled IBC tests

### CODE REVIEW

File: unittests/finality_proof.hpp [hunk 1]
```diff
@@ -168,21 +172,26 @@ namespace finality_proof {
 
          //skip this part on genesis
          if (!is_genesis){
+            parent_timestamp = timestamp;
             for (const auto& p : blocks_since_proposed_policy){
-               //under the happy path with strong QCs in every block, a policy becomes active `2 * eosio::testing::num_chains_to_final` blocks after being proposed
-               if (p.second.blocks_since_proposed == 2 * eosio::testing::num_chains_to_final){
+
+               //under the happy path with strong QCs in every block, a policy becomes active 6 blocks after being proposed
```
Comment (linh2931) [inline unittests/finality_proof.hpp:178]: If we don't use the constant `num_chains_to_final` in the comment, 6 should be 4

File: unittests/finality_proof.hpp [hunk 2]
```diff
@@ -168,21 +172,26 @@ namespace finality_proof {
...
-               if (p.second.blocks_since_proposed == 2 * eosio::testing::num_chains_to_final){
+
+               //under the happy path with strong QCs in every block, a policy becomes active 6 blocks after being proposed
+               if (p.second.blocks_since_proposed == 2 * eosio::testing::num_chains_to_final && p.first != active_finalizer_policy_digest){
                   active_finalizer_policy = p.second.policy;
                   active_finalizer_policy_digest = p.first;
                }
-               //under the happy path with strong QCs in every block, a policy becomes pending `num_chains_to_final` blocks after being proposed
-               else if (p.second.blocks_since_proposed == eosio::testing::num_chains_to_final){
+               //under the happy path with strong QCs in every block, a policy becomes pending 3 blocks after being proposed
```
Comment (linh2931) [inline unittests/finality_proof.hpp:183]: 3 should be 2

File: unittests/svnn_ibc_tests.cpp [hunk 3]
```diff
@@ -119,8 +115,8 @@ BOOST_AUTO_TEST_CASE(ibc_test) {} // TODO remove this and re-enable actual ibc_t
                   ("finality_mroot", block_3_result.finality_root)
                )
                ("qc", mvo()
-                  ("signature", block_5_result.qc_data.qc.value().active_policy_sig.sig.to_string())
-                  ("finalizers", active_finalizers_string(block_5_result)) 
+                  ("signature", block_4_result.qc_data.qc.value().active_policy_sig.sig.to_string())
```
Comment (linh2931) [inline unittests/svnn_ibc_tests.cpp:118]: Can you expand a little more about why/how block_3_result and block_4_result are used to prove block_2 in the comments above?

**Review by greg7mdp — APPROVED** [review]

File: unittests/finality_proof.hpp (hunk 1 above)
Comment (systemzax) [reply unittests/finality_proof.hpp]: Done

File: unittests/finality_proof.hpp (hunk 2 above)
Comment (systemzax) [reply unittests/finality_proof.hpp]: Done

File: unittests/svnn_ibc_tests.cpp (hunk 3 above)
Comment (linh2931) [reply unittests/svnn_ibc_tests.cpp:122]: Thanks! Your new comment is very clear.

**Review by linh2931 — APPROVED** [review]

**Comment by ericpassmore** [comment]
Note:start
group: STABILITY
category: TEST
summary: Update tests resolve merge conflicts from multiple PRs applied during switch to 2-chain.
Note:end

**Pull request merged**